core/client.go

```diff
@@ -3,24 +3,13 @@ package core
 import (
 	"fmt"
 
-	retryhttp "github.com/hashicorp/go-retryablehttp"
-	"github.com/tendermint/tendermint/rpc/client"
-	"github.com/tendermint/tendermint/rpc/client/http"
+	coregrpc "github.com/tendermint/tendermint/rpc/grpc"
 )
 
 // Client is an alias to Core Client.
-type Client = client.Client
+type Client = coregrpc.BlockAPIClient
 
-// NewRemote creates a new Client that communicates with a remote Core endpoint over HTTP.
+// NewRemote creates a new Client that communicates with a remote Core endpoint over gRPC.
 func NewRemote(ip, port string) (Client, error) {
-	httpClient := retryhttp.NewClient()
-	httpClient.RetryMax = 2
-	// suppress logging
-	httpClient.Logger = nil
-
-	return http.NewWithClient(
-		fmt.Sprintf("tcp://%s:%s", ip, port),
-		"/websocket",
-		httpClient.StandardClient(),
-	)
+	return coregrpc.StartBlockAPIGRPCClient(fmt.Sprintf("tcp://%s:%s", ip, port))
```

Comment:
Can we reuse the client across `state` and `core` modules?

Reply:
Yes. I'm planning to do it in a subsequent PR